src/main/java/pl/simpleascoding/tutoringplatform/api/TeacherController.java

```diff
@@ -0,0 +1,32 @@
+package pl.simpleascoding.tutoringplatform.api;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import pl.simpleascoding.tutoringplatform.dto.SignAsTeacherDTO;
+import pl.simpleascoding.tutoringplatform.dto.TeacherDTO;
+import pl.simpleascoding.tutoringplatform.service.teacher.TeacherService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("api/v1/teachers")
+@RequiredArgsConstructor
+public class TeacherController {
+
+    private final TeacherService teacherService;
+
+    @GetMapping("/all")
+    public ResponseEntity<List<TeacherDTO>> findAllTeachers() {
+        return new ResponseEntity<>(teacherService.findAllTeachers(), HttpStatus.OK);
+    }
```

Comment:
I would recommend implementing pagination on this endpoint. We don't know how many users will use our app and retrieving, for example, 10000 users would be very time consuming.
Also, I would leave the endpoint's address at simply `api/v1/teachers` instead of `api/v1/teachers/all`, but that is just my opinion and it doesn't really matter that much.

Reply:
I'll think about it. but to be honest, I don't think that we'll have 10000 users. If yes, we should use other framework to take care about databases. Hibernate is good and flexible but it's also made for small applications (like 2000/3000 users).

Reply:
About endpoint - 'all' refers to all teachers, and only all teachers.

Reply:
@Eukon, I do not agree. We will talk on the communicator on this issue.